index.js

```diff
@@ -2,12 +2,81 @@ const mongoose = require('mongoose');
 const Recipe = require('./models/Recipe.model'); // Import of the model Recipe from './models/Recipe.model.js'
 const data = require('./data.js'); // Import of the data from './data.js'
 
+const cake = {title: "Fondant", level: "Easy Peasy", cuisine: "italienne", creator: "Coline"};
+
 // Connection to the database "recipeApp"
-mongoose
-  .connect('mongodb://localhost/recipe-app-dev', {
-    useCreateIndex: true,
-    useNewUrlParser: true,
-    useUnifiedTopology: true
-  })
-  .then(x => console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`))
-  .catch(err => console.error('Error connecting to mongo', err));
+async function createRecipes() {
+  try {
+    const startDataBase = await mongoose.connect('mongodb://localhost/recipe-app-dev', {
+      useCreateIndex: true,
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    });
+    console.log("Promise1");
+    const createCake = await Recipe
+    .create(cake).catch(err => console.log(err)
+    );
+    console.log("Promise2");
+    const insertAll = await Recipe
+    .insertMany(data)
+    .catch(dbErr => console.log(`not inserted`, dbErr));
+    console.log("Promise3");
+    const updateCake = await Recipe
+    .updateOne({title: "Rigatoni alla Genovese"}, {duration: 100}, {new: true})
+    .catch(dbErr => console.log(`not inserted`, dbErr));
+    console.log("Promise4");
+    const deleteCake = await Recipe
+    .deleteOne({title: "Carrot Cake"})
+    .catch(dbErr => console.log(`not inserted`, dbErr));
+    console.log("Promise5");
+    const closeDB = mongoose
+    .connection.close()
+    .catch(err => console.log("There was an erros when closing the database"));
+    console.log("Promise6");
```

Comment:
Here you have to await the connection in order to have the "console.log("Promise 6") to happen.

```
const closeDB = await mongoose
    .connection.close()
    .catch(err => console.log("There was an erros when closing the database"));
    console.log("Promise6");

```